docs/contrib/06_new_project.md

```diff
@@ -0,0 +1,40 @@
+# Adding a project to pySDC (and automatic testing)
+
+When adding a new project to pySDC, it needs to be tested.
+In order to run those tests as fast as possible, a few steps are needed, that are given below.
+
+## Add a directory in 'pySDC/projects'
+
+First, create a new directory in `pySDC/projects` with the name of your project.
+The code of the new project should go into that newly created directory.
+
+## Add an environment-file
+
+In addition to the codde of your projects, an environment-file is needed.
+This includes the dependencies that are required for your project to run.
+The file needs to be named `env_update.yml` and needs to follow the structure shown below:
+
+```yaml
+name: pySDC
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - numpy
+```
+
+The list of dependencies can be extended as needed.
+The name should stay `pySDC`. The channels cover most of the usual packages.
+If a package is needed that cannot be found in those channels by conda (or mamba),
+please add the correct channel to the list.
+
+## Add tests to the project
+
+In order to automatically find the tests of your project, please add the tests to a subdirectory called `tests` in the directory of your project. Furthermore, the files should start with `test_` and the functions should also start with that. For more information see the documentation of pytest on [test discovery](<https://docs.pytest.org/en/8.2.x/explanation/goodpractices.html#tests-as-part-of-application-code>).
+
+## Add the project in the CI-file
+
+To run the tests of all projects in parallel, the projects are explicitly mentioned in the CI-file.
+In order to run the tests of your project, please add the name of your project **as your directory is named**
+in the [CI-File](./../../.github/workflows/ci_pipeline.yml)
+in the job `project_cpu_tests_linux` in the list `strategy/matrix/env`.
```

Comment:
Again, we need to make sure that this is correct in the end. If we decide to use the. pytest markers (I vote we do) we need to detail here how to add you own.

Reply:
I like the markers because they assign the tests to the respective environments in a very explicit way. But I can see that we disagree on this issue. @pancetta, please note your preference and we'll go with that.

Reply:
The markers were used originally because we wanted to filter the tests by a certain property. Adding new ones is pretty clumsy and forgetting to set a marker was also easily possible. So, if we don't need markers, we should get rid of them. But if we need to do other filter magic instead, we should keep them. So, what would happen if we drop markers?

Reply:
In any case, we should not mix markers and non-markers. If we decide to use markers, we should only use markers. If we decide to get rid of markers, we should get fully rid of them.

Reply:
Jakob currently set things up so that the project tests are in the project folders and the pipeline just runs all tests in a project folder with the environment file in the same folder. The remaining tests are still in the separate tests directory and use the markers `base`, `mpi4py`, ... 
So then you don't need to add markers when adding a new project. Fine. I agree that it's a bit simpler ;)

Reply:
Just to avoid confusion: We can get rid of all markers within the tests for projects. But currently we still use pytest-markers for all tests outside of projects. I wanted to explicitly state that as it slightly contradicts the statement of @pancetta about using markers everywhere or nowhere.
Is that fine with you, that we use them nowhere within the project-tests and everywhere outside of project-tests, @pancetta ?
I don't see a good way to get rid of the markers for the tests outside of projects, but don't think we need them within project-tests.

Reply:
Yeah, we did introduce inconsistencies here, too (see also my other comment below).